File: implementation/scipamato/scipamato-common-persistence-jooq/src/test/java/ch/difty/scipamato/common/persistence/AbstractFilterConditionMapperTest.java
package ch.difty.scipamato.common.persistence;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.List;

import org.jooq.Condition;
import org.junit.Test;

import ch.difty.scipamato.common.entity.filter.ScipamatoFilter;

public class AbstractFilterConditionMapperTest {

    private boolean filterCalled = false;

    private static class DummyFilter extends ScipamatoFilter {
        private static final long serialVersionUID = 1L;
    }

    private final AbstractFilterConditionMapper<DummyFilter> mapper = new AbstractFilterConditionMapper<>() {
        @Override
        protected void map(DummyFilter filter, List<Condition> conditions) {
            filterCalled = true;
        }
    };

    @Test
    public void mappingWithNullFilter_doesNotMap() {
        assertThat(mapper
            .map(null)
            .toString()).isEqualTo("1 = 1");
        assertThat(filterCalled).isFalse();
    }

    @Test
    public void mappingWithNonNullFilter_doesMap() {
        mapper.map(new DummyFilter());
        assertThat(filterCalled).isTrue();
    }

}

File: common/common-persistence-jooq/src/test/kotlin/ch/difty/scipamato/common/persistence/AbstractFilterConditionMapperTest.kt
package ch.difty.scipamato.common.persistence

import ch.difty.scipamato.common.entity.filter.ScipamatoFilter
import io.mockk.every
import io.mockk.mockk
import org.amshove.kluent.shouldBeEqualTo
import org.jooq.Condition
import org.junit.jupiter.api.Test

internal class AbstractFilterConditionMapperTest {

    private val condition = mockk<Condition>()

    private val filterConditionMapper = object : AbstractFilterConditionMapper<TestFilter>() {
        override fun internalMap(filter: TestFilter): List<Condition> = listOf(condition)
    }

    @Test
    fun withNullFilter() {
        val condition = filterConditionMapper.map(null)
        condition.toString() shouldBeEqualTo "1 = 1"
    }

    @Test
    fun foo() {
        val cond = "cond"
        every { condition.toString() } returns cond

        val condition = filterConditionMapper.map(TestFilter())

        condition.toString() shouldBeEqualTo cond
    }
}

class TestFilter : ScipamatoFilter()
